pkg/config/mesh/mesh.go

```diff
@@ -40,63 +40,144 @@ var (
 
 // DefaultProxyConfig for individual proxies
 func DefaultProxyConfig() meshconfig.ProxyConfig {
+	if TestMode {
+		return meshconfig.ProxyConfig{
+			ConfigPath:             constants.ConfigPathDir,
+			BinaryPath:             constants.BinaryPathFilename,
+			ServiceCluster:         constants.ServiceClusterName,
+			DrainDuration:          types.DurationProto(45 * time.Second),
+			ParentShutdownDuration: types.DurationProto(60 * time.Second),
+			DiscoveryAddress:       constants.DiscoveryPlainAddress,
+			StatsdUdpAddress:       "",
+			EnvoyMetricsService:    &meshconfig.RemoteService{Address: ""},
+			EnvoyAccessLogService:  &meshconfig.RemoteService{Address: ""},
+			ProxyAdminPort:         15000,
+			ControlPlaneAuthPolicy: meshconfig.AuthenticationPolicy_NONE,
+			CustomConfigFile:       "",
+			Concurrency:            0,
+			StatNameLength:         189,
+			Tracing:                nil,
+			StatusPort:             15020,
+		}
+	}
+	// TODO: include revision based on REVISION env
+	// TODO: set default namespace based on POD_NAMESPACE env
 	return meshconfig.ProxyConfig{
+		// missing: ConnectTimeout: 10 * time.Second,
 		ConfigPath:             constants.ConfigPathDir,
-		BinaryPath:             constants.BinaryPathFilename,
 		ServiceCluster:         constants.ServiceClusterName,
 		DrainDuration:          types.DurationProto(45 * time.Second),
 		ParentShutdownDuration: types.DurationProto(60 * time.Second),
-		DiscoveryAddress:       constants.DiscoveryPlainAddress,
-		StatsdUdpAddress:       "",
-		EnvoyMetricsService:    &meshconfig.RemoteService{Address: ""},
-		EnvoyAccessLogService:  &meshconfig.RemoteService{Address: ""},
 		ProxyAdminPort:         15000,
+		Concurrency:            2,
 		ControlPlaneAuthPolicy: meshconfig.AuthenticationPolicy_NONE,
-		CustomConfigFile:       "",
-		Concurrency:            0,
-		StatNameLength:         189,
-		Tracing:                nil,
-		StatusPort:             15020,
+		DiscoveryAddress:       "istiod.istio-system.svc:15012",
+		Tracing: &meshconfig.Tracing{
+			Tracer: &meshconfig.Tracing_Zipkin_{
+				Zipkin: &meshconfig.Tracing_Zipkin{
+					Address: "zipkin.istio-system:9411",
+				},
+			},
+		},
+
+		// Code defaults
+		BinaryPath:            constants.BinaryPathFilename,
+		StatsdUdpAddress:      "",
+		EnvoyMetricsService:   &meshconfig.RemoteService{Address: ""},
+		EnvoyAccessLogService: &meshconfig.RemoteService{Address: ""},
+		CustomConfigFile:      "",
+		StatNameLength:        189,
+		StatusPort:            15020,
 	}
 }
 
+var (
+	TestMode = false
+)
+
 // DefaultMeshConfig configuration
 func DefaultMeshConfig() meshconfig.MeshConfig {
 	proxyConfig := DefaultProxyConfig()
+	if TestMode {
```

Comment:
What's this for？

Reply:
We have a lot of tests (mostly unit) that have checks assuming the old defaults. Will take some time to update all of them - this allows fixing the defaults for prod, without having to wait until we update all tests (which is more work than I have time for). See the failures - I'm planning to set it in the test startup.

A hack - but it seems more useful to have good prod-matching defaults then wait forever for tests to use real values.